src/components/Patient/PatientRegistration.tsx

```diff
@@ -0,0 +1,1076 @@
+    : t("update_patient_details");
+
+  const errors = { ...feErrors, ...createPatientMutation.error };
+
+  const fieldProps = (field: keyof typeof form) => ({
+    value: form[field] as string,
+    onChange: (e: React.ChangeEvent<HTMLInputElement>) =>
+      setForm((f) => ({
+        ...f,
+        [field]: e.target.value === "" ? undefined : e.target.value,
+      })),
+  });
+
+  const selectProps = (field: keyof typeof form) => ({
+    value: (form[field] as string)?.toString(),
+    onValueChange: (value: string) =>
+      setForm((f) => ({ ...f, [field]: value })),
+  });
+
+  const handleDialogClose = (action: string) => {
+    if (action === "transfer") {
+      setShowTransferDialog(true);
+    } else if (action === "back") {
+      setShowTransferDialog(false);
+    } else {
+      setSuppressDuplicateWarning(true);
+      setShowTransferDialog(false);
+    }
+  };
+
+  const handleFormSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    const errors: Record<string, string[]> = {};
+    const requiredFields: Array<keyof typeof form> = [
+      "name",
+      "phone_number",
+      "emergency_phone_number",
+      "gender",
+      "blood_group",
+      ageDob === "dob" ? "date_of_birth" : "year_of_birth",
+      "pincode",
+      "nationality",
+      "address",
+      "permanent_address",
+    ];
+
+    if (form.nationality === "India") {
+      requiredFields.push("state", "district", "local_body");
+    }
+
+    requiredFields.forEach((field) => {
+      if (!form[field]) {
+        errors[field] = errors[field] || [];
+        errors[field].push(`This field is required`);
+      } else if (
+        ageDob === "dob" &&
+        field === "date_of_birth" &&
+        !/^(19[0-9]{2}|20[0-9]{2}|2100)-(0?[1-9]|1[0-2])-(0?[1-9]|[12]\d|3[01])$/.test(
+          form[field],
+        )
+      ) {
+        errors[field] = errors[field] || [];
+        errors[field].push(t("invalid_date_format", { format: "DD-MM-YYYY" }));
+      } else if (
+        (field === "phone_number" || field === "emergency_phone_number") &&
+        form[field]?.length < 13
+      ) {
+        errors[field] = errors[field] || [];
+        errors[field].push(t("phone_number_min_error"));
+      }
+    });
+
+    if (Object.keys(errors).length > 0) {
+      setFeErrors(errors);
+      Notification.Error({
+        msg: t("please_fix_errors"),
+      });
+    } else {
+      patientId
+        ? updatePatientMutation.mutate()
+        : createPatientMutation.mutate();
+    }
+  };
+
+  const [debouncedNumber, setDebouncedNumber] = useState<string>();
+
+  useEffect(() => {
+    const handler = setTimeout(() => {
+      if (!patientId || patientQuery.data?.phone_number !== form.phone_number) {
+        setSuppressDuplicateWarning(false);
+      }
+      setDebouncedNumber(form.phone_number);
+    }, 500);
+
+    return () => {
+      clearTimeout(handler);
+    };
+  }, [form.phone_number]);
+
+  const patientPhoneSearch = useQuery({
```

Comment:
I think a better workflow would be to freeze the phoneNumber field in the PatientRegister Form, given that the user has already gone through the search and find phase.

Maybe file an issue

Reply:
Are you suggesting to freeze the input field with a number if specified in query params? 

flow will be : user searches for phone number xyz, finds no patients, clicks on create patient which takes the user to /patient-register?number=xyz with a frozen number field

- What happens if the user had initially searched for the wrong phone number, and now wants to change it midway through filling the form?
- It would still need a patient phone search for users that are created without searching for the number first, so functionality would not change.